src/statuses.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Statuses for the Charmed Etcd Operator.
+
+This module defines various status enums that represent the state of the charm,
+"""
+
+from enum import Enum
+
+from data_platform_helpers.advanced_statuses.models import StatusObject
+
+
+class CharmStatuses(Enum):
+    """Collection of possible statuses for the charm."""
+
+    ACTIVE_IDLE = StatusObject(status="active", message="")
+    NO_PEER_RELATION = StatusObject(status="maintenance", message="no peer relation available")
+    PEER_URL_NOT_SET = StatusObject(status="maintenance", message="peer-url not set")
+
+
+class BackupStatuses(Enum):
+    """Collection of backup related statuses."""
+
+    BACKUP_IN_PROGRESS = StatusObject(status="maintenance", message="Creating database backup...")
+    BACKUP_S3_PARAMETERS_MISSING = StatusObject(
+        status="blocked", message="Missing or invalid s3 credentials"
+    )
+    BACKUP_AZURE_PARAMETERS_MISSING = StatusObject(
+        status="blocked", message="Missing or invalid azure credentials"
+    )
+    RESTORE_FAILED = StatusObject(status="blocked", message="failed to restore backup")
+    RESTORE_VERIFICATION_FAILED = StatusObject(
+        status="blocked",
+        message="Restore verification failed - etcd cluster still running, restore cancelled, check debug-log",
+    )
+    RESTORE_IN_PROGRESS = StatusObject(
+        status="maintenance", message="Database restore is in progress"
+    )
+    RESTORE_UNHEALTHY = StatusObject(
+        status="blocked",
+        message="cluster unhealthy after restoring backup - check debug-log",
+    )
+    OBJECT_STORAGE_CONFLICT = StatusObject(
+        status="blocked", message="Azure and S3 storages configured - please remove one"
+    )
+
+
+class ClusterStatuses(Enum):
+    """Collection of cluster related statuses."""
+
+    CLUSTER_INITIALIZING = StatusObject(
+        status="maintenance", message="Initializing etcd cluster..."
+    )
+    CLUSTER_FAILED = StatusObject(
+        status="blocked",
+        message="Cluster failure - majority of cluster members lost",
+        running="async",
+        action="rebuild-cluster",
```

Comment:
It's maybe not very clear but you can be a bit more verbose on the "action" field, because it represents an action to take (as an operator). It could be to run a juju action, or check some configuration, etc.

```suggestion
        action="Run action rebuild-cluster",
```